### DISCUSSION TITLE
New methods for Bezier curve - length and value at arc length

### DISCUSSION THREAD

**Discussion by achirita** [jscad/OpenJSCAD.org#1215] [Ideas]
Hello,

I'm working on a JSCAD project and noticed there are no methods for measuring the length of an Bezier curve as well as a way of getting the value at a specific arc length rather than a time value.
Both of these would be useful in a wide range of projects and I would be happy to contribute a pull request.
Length measurement is a complicated topic but we could use the naïve approach of sampling the curve at multiple points then adding up all the distances. The more points you use the better the accuracy gets. Three.js uses the same approach and much of my code is adapted from there.
If you're interested, let me know.

**Comment by z3dev** [comment]
@achirita thanks for using  JSCAD. And of course, Please feel free to contribute.

Just a few suggestions on the pull requests... Keep changes as small as possible as that make reviews easier... Please add a simple test for the changes.

By the way, we are looking to add 'arc' to the curves functions. If interested then this would be a nice addition as well. 😃

**Comment by achirita** [comment]
New methods for Bezier curves were added to the master branch and will probably be released soon.
